src/components/Dashboard/NextMealScheduled.tsx

```diff
@@ -1,19 +1,31 @@
+"use client";
+
 import Image from "next/image";
 import Link from "next/link";
 
+import { getUpcomingFoodEventDetails } from "@/app/get-food-ticket/actions";
 import CalendarIcon from "@/images/dashboard/Calendar.png";
+import { useQuery } from "@tanstack/react-query";
 
+import { useUser } from "../contexts/UserContext";
 import Card from "./Card";
 
 export default function NextMealScheduled() {
   const href = "#";
-  const mealTime = new Date().toLocaleTimeString("en-US", {
-    hour: "numeric",
-    minute: "numeric",
+  const userId = useUser().currentUser.username as string;
+  const { data, isFetching } = useQuery({
+    initialDataUpdatedAt: 0,
+    queryKey: ["User", userId],
```

Comment:
This query key is not accurate. You are not loading the user but loading time slot instead. We should name it as such also it doesn't override the user fetches with different data.